sound/soc/sof/ipc4-pcm.c

```diff
@@ -89,10 +89,24 @@ sof_ipc4_add_pipeline_to_trigger_list(struct snd_sof_dev *sdev, int state,
 				pipe_widget->instance_id;
 		break;
 	case SOF_IPC4_PIPE_PAUSED:
-		/* Pause the pipeline only when its started_count is 1 more than paused_count */
-		if (spipe->paused_count == (spipe->started_count - 1))
-			trigger_list->pipeline_ids[trigger_list->count++] =
-				pipe_widget->instance_id;
+		switch (cmd) {
+		case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
+		case SNDRV_PCM_TRIGGER_SUSPEND:
+		case SNDRV_PCM_TRIGGER_STOP:
+			/* Pause the pipeline if its started_count is 1 more than paused_count */
+			if (spipe->paused_count == (spipe->started_count - 1))
+				trigger_list->pipeline_ids[trigger_list->count++] =
+					pipe_widget->instance_id;
+			break;
+		case SNDRV_PCM_TRIGGER_START:
```

Comment:
not sure why TRIGGER_PAUSE_RELEASE is not handled here, is this intentional? If yes a comment would help.

Reply:
state can never be PAUSED when we're doing a release